### DISCUSSION TITLE
UserTokenRequestsFetcher and bytes

### DISCUSSION THREAD

**Discussion by apwob1** [joalla/discogs_client#61] [Support] [answered]
Hi,

First of all, thanks for maintaining!

I'm trying to use the UserTokenRequestsFetcher class to fetch all of my Discogs Lists (and later, their Items). Ultimately I'm trying to push this data into a local database, and so need to process the data as such.

As per the documentation, it returns a bytes object, but being fairly new to Python, I'm struggling to move forward from that point. I've got it converted to a dictionary, but then it doesn't behave as I would expect (throwing errors when trying to convert it from there into a pandas DataFrame.

Appreciate this is more of a me-issue than a library-issue, but just wondered if you there was an expected set of next steps to take with results from the Fetcher classes? Is there another part of this library that could help me get what I need from the API?

Cheers in advance!

**Comment by AnssiAhola** [comment]
Hey @apwob1

Im not sure I get what you are trying to do 😅
Theres a `Client` class that you can use to make calls to the api.

You can get a list by id using `client.list('id')`
Or your lists by `client.user('yourusername').lists`

Have you read our new docs? https://python3-discogs-client.readthedocs.org/

Hope that helped atleast in some way, cheers.

**Reply by apwob1** [reply]
Hi, yep as mentioned in my post I've read the docs - retrieving my lists isn't a problem, more what is the expected interaction with the bytes object that gets returned from the class I mentioned.

**Reply by AnssiAhola** [reply]
Hey. Sorry completely forgot about this 😅 .
If you could, a code example would help a lot so I can reproduce this.
That is, if you haven't solved it already 😄

**Reply by apwob1** [reply]
Hi, thanks for replying, below is what I'm dealing with;

```
d = dc.Client('MyApp/0.1', user_token=ra.key)

f = dc.fetchers.UserTokenRequestsFetcher(ra.key)

result = f.fetch(client=d, method='GET', url='https://api.discogs.com/users/*****/lists')[0]
```

I get this far, and `result` is a bytes object. I'm trying to get the data into a pandas dataframe, but it won't load from this, nor from a dict I converted it to using json.loads.

I'm sure it's something fairly obvious I just can't seem to google the right thing to answer my question!

Thanks.

**Reply by JOJ0** [reply]
Hi😀, not sure if relevant with your issue but which python version are you using?

**Reply by JOJ0** [reply]
Ah, ok now I took the time to read your initial post again and finally understand what you are trying to do. You want to instantiate the UserTokenRequestsFetcher yourself. Then you pass in a url and try to fetch from it. I never tried this and I don't think you should do that / want to do that. This is not the intended way of interacting with discogs_client. As Anssi already pointed out: Please just use the list methods of client and user instance and then let's work from there. Do you have the same issues when trying to add to pandas df?
Hope that helps! :-) Get back to us with any progress. thanks!

**Reply by JOJ0** [reply]
And while looking at our docs I find that the list(s) methods are not documented. Unfortunately our fetching data section is lacking an example for that: https://python3-discogs-client.readthedocs.io/en/latest/fetching_data.html

I only find this, which is a rather poor description of what the method does 💯 haha: https://python3-discogs-client.readthedocs.io/en/latest/discogs_client.html#discogs_client.models.User.lists

This is, well, not too bad in comparision ;-))) https://python3-discogs-client.readthedocs.io/en/latest/discogs_client.html#discogs_client.client.Client.list

I hope we find the time to improve docs on this end, sorry for that in the meantime, we'll figure your issue out without docs ;-)

**Reply by JOJ0** [reply, marked as answer]
And to be "superclear": To properly authenticate using the UserTokenRequestsFetcher only the first line of your code should be necessary. E.g as described here: https://python3-discogs-client.readthedocs.io/en/latest/authentication.html#user-token-authentication
